confgenerator/confgenerator.go

```diff
@@ -209,7 +209,7 @@ func (l *Logging) generateFluentbitComponents(userAgent string, hostInfo *host.I
 					}
 					components = append(components, processor.Components(tag, strconv.Itoa(i))...)
 				}
-				components = append(components, setLogNameComponents(tag, rID, receiver.Type())...)
+				components = append(components, setLogNameComponents(tag, rID, receiver.Type(), hostInfo.Hostname)...)
```

Comment:
hostname isn't necessarily the same as the instance name - the resource name has to come from the GCE metadata server

Reply:
I went back to explore the initial discussion ( b/169784211#comment18 ) with @igorpeshansky and @qingling128 about this task and realized why it was ok to use `hostInfo.Hostname`. 

The [value that is used](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/dce3e4b53ec4ff35e72d3c5fac1cf38428048e9d/lib/fluent/plugin/out_google_cloud.rb#L1126) in `google-fluentd` (which this feature wants to mirror) is actually the [`Socket.gethostname`](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/lib/fluent/plugin/common.rb#L196
) which is equivalent of getting `os.Hostname` in go with `gopsutil`.